src/analyses/baseUtil.ml

```diff
@@ -0,0 +1,19 @@
+open Prelude.Ana
+open GobConfig
+module Q = Queries
+
+let is_global (a: Q.ask) (v: varinfo): bool =
+  v.vglob || match a (Q.MayEscape v) with `MayBool tv -> tv | _ -> false
```

Comment:
Should this not default to `true` in case `threadEscape` is off?

Reply:
That would be (more) sound indeed. The consequence of that is all variables become global, which is hardly ever the desired outcome. I guess the past intent was that disabling escape analysis shouldn't have that outcome. I can make this change and add a warning about it too.

This also requires updating a bunch of tests to make them actually list escape analysis.

Reply:
I just committed the fix, but that just caused all partitioned array tests to fail. I guess it tries to use `is_global` with some degenerate `ctx` somewhere?

Reply:
>  I can make this change and add a warning about it too.

Yes, I think that is a good idea!

Reply:
> I guess it tries to use is_global with some degenerate ctx somewhere?
I'll take a look, that is my code after all :smile:

Reply:
>  should we move this fix to a separate branch and PR?

Yes, let's do that!

Reply:
I'll revert them here and cherry-pick to new branch then.